Code/Synnotech.FluentProcesses/Extensions.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Diagnostics;
+using Light.GuardClauses;
+
+namespace Synnotech.FluentProcesses;
+
+/// <summary>
+/// Provides extensions methods for processes.
+/// </summary>
+public static class Extensions
+{
+    /// <summary>
+    /// Creates a deep clone of the <paramref name="processStartInfo" /> instance.
+    /// Environment variables will only be copied when
+    /// <paramref name="copyEnvironment" /> is set to true. This is done to
+    /// prevent loading environment variables unless absolutely necessary
+    /// (the environment variables are loaded when
+    /// <see cref="ProcessStartInfo.Environment" /> is accessed for the first time).
+    /// </summary>
+    /// <param name="processStartInfo">The instance that should be cloned.</param>
+    /// <param name="copyEnvironment">
+    /// The value indicating whether the environment variables associated with the
+    /// process start info instance should be copied, too. The default value is false.
+    /// </param>
+    /// <exception cref="ArgumentNullException">Thrown when <paramref name="processStartInfo" /> is null.</exception>
+    public static ProcessStartInfo Clone(this ProcessStartInfo processStartInfo, bool copyEnvironment = false)
+    {
+        processStartInfo.MustNotBeNull();
+
+        var clone = new ProcessStartInfo
+        {
+            Arguments = processStartInfo.Arguments,
+            CreateNoWindow = processStartInfo.CreateNoWindow,
+            Verb = processStartInfo.Verb,
+            ErrorDialog = processStartInfo.ErrorDialog,
+            FileName = processStartInfo.FileName,
+            UserName = processStartInfo.UserName,
+            WindowStyle = processStartInfo.WindowStyle,
+            WorkingDirectory = processStartInfo.WorkingDirectory,
+            RedirectStandardError = processStartInfo.RedirectStandardError,
+            RedirectStandardInput = processStartInfo.RedirectStandardInput,
+            RedirectStandardOutput = processStartInfo.RedirectStandardOutput,
+            StandardErrorEncoding = processStartInfo.StandardErrorEncoding,
+            StandardOutputEncoding = processStartInfo.StandardOutputEncoding,
+            UseShellExecute = processStartInfo.UseShellExecute,
+            ErrorDialogParentHandle = processStartInfo.ErrorDialogParentHandle,
+        };
+
+#if NET6_0
+        if (OperatingSystem.IsWindows())
+            CopyWindowsSpecificProperties(processStartInfo, clone);
+#else
+        CopyWindowsSpecificProperties(processStartInfo, clone);
```

Comment:
If the target is .NET Standard 2.0, couldn't we run on Linux?

Reply:
Only if used with Mono. However, there is no .NET Standard API to check which platform you are running on, as far as I remember. But we could make this configurable.

Reply:
Hold on - I just found this:
https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.runtimeinformation.osdescription?view=net-6.0#system-runtime-interopservices-runtimeinformation-osdescription